In [9]:
import json 
import numpy as np
from collections import Counter

with open('retrieval_base_layer7_bert-base-uncased_layer1_p4_k100.json', 'r') as f:
    encodecmae_bert = json.load(f)

with open('retrieval_wav2vec2_layer6_bert-base-uncased_layer0_p4_k100.json', 'r') as f:
    w2v_bert = json.load(f)

# Palabras bien devueltas

In [17]:
keys_test = [encodecmae_bert['all_values'][i] for i in encodecmae_bert['rows_deleted']]
total = len(keys_test)
for i in range(total):
    word = keys_test[i]
    k_1 = encodecmae_bert['retrieval_indices'][i][0]
    if word == encodecmae_bert['keys_anchors'][k_1]:
        print(f"word: {word} retrieval:{encodecmae_bert['keys_anchors'][k_1]}")    
        print('---------')


word: of retrieval:of
---------
word: we retrieval:we
---------
word: illustration retrieval:illustration
---------
word: was retrieval:was
---------
word: me retrieval:me
---------
word: in retrieval:in
---------
word: river retrieval:river
---------
word: so retrieval:so
---------
word: keep retrieval:keep
---------
word: spirit retrieval:spirit
---------
word: of retrieval:of
---------
word: walked retrieval:walked
---------
word: the retrieval:the
---------
word: we retrieval:we
---------
word: that retrieval:that
---------
word: have retrieval:have
---------
word: do retrieval:do
---------
word: some retrieval:some
---------
word: not retrieval:not
---------
word: the retrieval:the
---------
word: these retrieval:these
---------
word: things retrieval:things
---------
word: the retrieval:the
---------
word: give retrieval:give
---------
word: got retrieval:got
---------
word: et retrieval:et
---------
word: the retrieval:the
---------
word: food retrieval:food
---------
word: in r

In [18]:
keys_test = [w2v_bert['all_values'][i] for i in w2v_bert['rows_deleted']]
total = len(keys_test)
for i in range(total):
    word = keys_test[i]
    k_1 = w2v_bert['retrieval_indices'][i][0]
    if word == w2v_bert['keys_anchors'][k_1]:
        print(f"word: {word} retrieval:{w2v_bert['keys_anchors'][k_1]}")    
        print('---------')

word: some retrieval:some
---------
word: this retrieval:this
---------
word: of retrieval:of
---------
word: fire retrieval:fire
---------
word: we retrieval:we
---------
word: never retrieval:never
---------
word: illustration retrieval:illustration
---------
word: was retrieval:was
---------
word: me retrieval:me
---------
word: probably retrieval:probably
---------
word: in retrieval:in
---------
word: garden retrieval:garden
---------
word: pocket retrieval:pocket
---------
word: voice retrieval:voice
---------
word: an retrieval:an
---------
word: keep retrieval:keep
---------
word: that retrieval:that
---------
word: river retrieval:river
---------
word: for retrieval:for
---------
word: during retrieval:during
---------
word: so retrieval:so
---------
word: their retrieval:their
---------
word: keep retrieval:keep
---------
word: country retrieval:country
---------
word: turban retrieval:turban
---------
word: like retrieval:like
---------
word: tore retrieval:tore
---------
wo

# Palabras mal devueltas

Consideramos palabras mal devueltas si no aparece la palabra que deberia en el top 5

In [19]:
keys_test = [encodecmae_bert['all_values'][i] for i in encodecmae_bert['rows_deleted']]
total = len(keys_test)
words_encodec = []
count = 0
for i in range(total):
    word = keys_test[i]
    k_1 = encodecmae_bert['retrieval_indices'][i]
    if not any(encodecmae_bert['keys_anchors'][index] == word for index in k_1 if index < len(encodecmae_bert['keys_anchors'])):
        word_retrieval =  [encodecmae_bert['keys_anchors'][k_1] for k_1 in encodecmae_bert['retrieval_indices'][i]]
        words_encodec.append([word, word_retrieval])
        count +=1 
        print(f"word: {word} retrieval:{word_retrieval}, indice {i}")    
        print('---------')
print(f'porcentaje de no devueltas: {count/total}')        
print(f'palabras: {count}') 

word: some retrieval:['insist', 'sooner', 'was', 'miss', 'was'], indice 0
---------
word: this retrieval:['the', 'the', 'the', 'the', 'the'], indice 1
---------
word: fire retrieval:['are', 'are', 'are', 'are', 'are'], indice 3
---------
word: never retrieval:['am', 'am', 'am', 'am', 'am'], indice 5
---------
word: excitement retrieval:['nile', 'wonder', 'autumn', 'general', 'twilight'], indice 9
---------
word: probably retrieval:['wildly', 'emperors', 'little', 'solved', 'little'], indice 10
---------
word: cross retrieval:['karl', 'karl', 'hot', 'crowd', 'casus'], indice 12
---------
word: and retrieval:['in', 'in', 'in', 'in', 'in'], indice 13
---------
word: no retrieval:['mode', 'mode', 'mode', 'mode', 'mode'], indice 15
---------
word: pocket retrieval:['back', 'back', 'up', 'up', 'up'], indice 16
---------
word: voice retrieval:['oil', 'war', 'tours', 'holgrave', 'will'], indice 17
---------
word: an retrieval:['in', 'in', 'in', 'in', 'in'], indice 18
---------
word: keep retri

In [20]:
keys_test = [w2v_bert['all_values'][i] for i in w2v_bert['rows_deleted']]
total = len(keys_test)
words_w2v = []
count = 0
for i in range(total):
    word = keys_test[i]
    k_1 = w2v_bert['retrieval_indices'][i]
    if not any(w2v_bert['keys_anchors'][index] == word for index in k_1 if index < len(w2v_bert['keys_anchors'])):
        word_retrieval =  [w2v_bert['keys_anchors'][k_1] for k_1 in w2v_bert['retrieval_indices'][i]]
        words_w2v.append([word, word_retrieval])
        count +=1 
        print(f"word: {word} retrieval:{word_retrieval}, indice {i}")    
        print('---------')
print(f'porcentaje de no devueltas: {count/total}')              
print(f'palabras: {count}') 

word: cross retrieval:['across', 'across', 'across', 'across', 'across'], indice 12
---------
word: no retrieval:['now', 'now', 'now', 'now', 'now'], indice 15
---------
word: recipe retrieval:['stew', 'christie', 'fish', 'he', 'he'], indice 20
---------
word: wielded retrieval:['sealed', 'will', 'will', 'will', 'will'], indice 34
---------
word: drouet retrieval:['herbert', 'ardour', 'herbert', 'ardor', 'bread'], indice 37
---------
word: panada retrieval:['canada', 'add', 'atlanta', 'add', 'pleroma'], indice 44
---------
word: quarters retrieval:['quarter', 'quarter', 'quarter', 'corner', 'rope'], indice 50
---------
word: olden retrieval:['old', 'old', 'old', 'old', 'old'], indice 54
---------
word: call retrieval:['called', 'called', 'called', 'called', 'called'], indice 55
---------
word: venus retrieval:['v', 'v', 'main', 'main', 'phoebe'], indice 61
---------
word: while retrieval:['wild', 'wild', 'wild', 'wild', 'wild'], indice 63
---------
word: edition retrieval:['tuition', '

¿Cuántas palabras hay "mal devueltas" en cada uno?

In [21]:
len(keys_test)

7035

In [22]:
len(words_w2v)

1803

In [23]:
len(words_encodec)

4482

In [25]:
words_w2v_counter = [item[0] for item in words_w2v]
words_encodec_counter = [item[0] for item in words_encodec]

# Count each word in both lists
counter_w2v = Counter(words_w2v_counter)
counter_encodec = Counter(words_encodec_counter)

# Find the intersection of words in both lists
common_words = set(words_w2v_counter) & set(words_encodec_counter)

# Print the results
print("Words in words_w2v:", len(counter_w2v))
print("Words in words_encodec:", len(counter_encodec))
print("Common words:", common_words)
print("Number of common words:", len(common_words))

Words in words_w2v: 1523
Words in words_encodec: 2300
Common words: {'split', 'lacking', 'challoner', 'steely', 'newspaper', 'spend', 'remote', 'christi', 'final', 'underneath', 'gascon', 'captives', 'appreciated', 'manager', 'uproarious', 'observe', 'thirst', 'trifling', 'blessed', 'hypocrites', 'written', 'thrusting', 'blind', 'inclined', 'discordant', 'breed', 'necromancy', 'moaning', 'misery', 'labours', 'station', 'poems', 'sun', 'dismissed', 'preoccupied', 'grandmother', 'scarf', 'devil', 'servant', 'reserved', 'stirring', 'exalted', 'interview', 'unmeaning', 'physical', 'cross', 'permanent', 'generation', 'bed', 'acquaintance', 'interior', 'pell', 'sadness', 'enterprise', 'guess', 'oh', 'everyday', 'us', 'breaking', 'routine', 'wagons', 'he', 'phil', 'moccasins', 'above', 'rushed', 'lengths', 'greeting', 'morality', 'edifice', 'magnetism', 'shaken', 'gloat', 'translated', 'trubner', 'lassitude', 'mildly', 'yield', 'dangerous', 'lovely', 'marked', 'mask', 'spoken', 'lattimer', 'h

Veamos las palabras que habia que devolver cuántas apariciones tiene

In [26]:
with open('../../words_counts.json', 'r') as json_file:
    words_count = json.load(json_file)

In [27]:
common_word_counts = {word: words_count.get(word, 0) for word in keys_test}
count_one_value = sum(1 for count in common_word_counts.values() if count == 1)

print("Number of words with a count of 1 on key_test:", count_one_value)

Number of words with a count of 1 on key_test: 430


In [28]:
common_word_counts = {word: words_count.get(word, 0) for word in common_words}
count_one_value = sum(1 for count in common_word_counts.values() if count == 1)

print("Number of words with a count of 1 on common_words:", count_one_value)

Number of words with a count of 1 on common_words: 430


Hay 430 palabras que no habia visto el encoder ¿Qué devuelve para esas palabras?

In [29]:
count = 0
words_encodec_unique = []
for i in range(len(words_encodec)):
    word = words_encodec[i][0]
    if words_count[word] == 1:
        words_encodec_unique.append(words_encodec[i])
        count += 1
        print(f"word: {word} retrieval:{words_encodec[i][1]}, indice {i}")    

word: wielded retrieval:['little', 'little', 'little', 'little', 'little'], indice 24
word: performance retrieval:['from', 'from', 'from', 'from', 'from'], indice 47
word: charwoman retrieval:['him', 'hour', 'him', 'him', 'him'], indice 52
word: darker retrieval:['our', 'our', 'our', 'our', 'our'], indice 64
word: appointment retrieval:['suppressed', 'reception', 'post', 'bright', 'return'], indice 68
word: woven retrieval:['warm', 'warm', 'warm', 'ball', 'mood'], indice 71
word: bustle retrieval:['up', 'up', 'up', 'up', 'up'], indice 77
word: distinctively retrieval:['ceiling', 'schloss', 'pleasing', 'fully', 'dainty'], indice 93
word: ampler retrieval:['them', 'them', 'them', 'them', 'them'], indice 95
word: pales retrieval:['hoe', 'rows', 'daylight', 'have', 'veil'], indice 106
word: ancestral retrieval:['sunrise', 'known', 'founded', 'man', 'loves'], indice 108
word: administrator retrieval:['admire', 'wagon', 'reeling', 'machine', 'green'], indice 117
word: lantern retrieval:['rei

In [30]:
print(count)

430


In [31]:
count = 0
words_w2v_unique = []
for i in range(len(words_w2v)):
    word = words_w2v[i][0]
    if words_count[word] == 1:
        words_w2v_unique.append(words_w2v[i])
        count += 1
        print(f"word: {word} retrieval:{words_w2v[i][1]}, indice {i}")    

word: wielded retrieval:['sealed', 'will', 'will', 'will', 'will'], indice 3
word: performance retrieval:['reform', 'for', 'for', 'for', 'for'], indice 12
word: charwoman retrieval:['bun', 'them', 'reward', 'them', 'them'], indice 13
word: darker retrieval:['dark', 'dark', 'dark', 'dark', 'dark'], indice 19
word: appointment retrieval:['drainage', 'marriage', 'complained', 'bun', 'meantime'], indice 20
word: woven retrieval:['womb', 'wave', 'rudely', 'wave', 'ward'], indice 21
word: bustle retrieval:['bells', 'buffalo', 'robust', 'mall', 'solve'], indice 22
word: distinctively retrieval:['indistinct', 'existing', 'indistinctly', 'instinctive', 'league'], indice 26
word: ampler retrieval:['man', 'man', 'man', 'man', 'man'], indice 27
word: pales retrieval:['rows', 'of', 'of', 'of', 'of'], indice 30
word: ancestral retrieval:['senses', 'external', 'thence', 'mosaic', 'remonstrances'], indice 31
word: administrator retrieval:['witted', 'reduce', 'misuse', 'limit', 'mating'], indice 36
wor

In [32]:
import pandas as pd
words_w2v_counter = [item[0] for item in words_w2v_unique]
words_encodec_counter = [item[0] for item in words_encodec_unique]

# Find the intersection of words in both lists
common_words = set(words_w2v_counter) & set(words_encodec_counter)

# Creating DataFrame with word lists
data = {
    'word': list(common_words),
    'encodec': [next(item[1] for item in words_encodec_unique if item[0] == word) for word in common_words],
    'w2v': [next(item[1] for item in words_w2v_unique if item[0] == word) for word in common_words]
}

df_unique = pd.DataFrame(data)

df_unique.to_csv('retrieval_unique.csv')


In [33]:
df_unique[:150]

word                                      encodec  \
0     physicians                         [us, us, us, us, us]   
1        lacking             [load, angel, aid, narrow, rain]   
2      challoner                    [are, are, are, are, are]   
3      falsehood       [falls, fall, soils, decision, flowed]   
4     reconciled         [nile, skies, rein, thereof, mosaic]   
..           ...                                          ...   
145       citron     [should, should, should, should, should]   
146    requested     [bar, drizzle, ushered, papers, devoted]   
147  approvingly   [aid, ability, longingly, borrowing, line]   
148       gaping                       [be, be, be, king, be]   
149     diligent  [admitted, meant, notice, length, eleventh]   

                                                   w2v  
0                 [ewes, ewes, physician, dresses, us]  
1             [mating, making, making, making, making]  
2         [merlin, mother, mother, amount, memorandum]  
3                 [falls, faults, souls, solve, souls]  
4                  [collar, thaw, stair, sack, arouse]  
..                                                 ...  
145                [serge, sense, sense, sense, sense]  
146       [quest, quest, quest, affecting, suppressed]  
147  [longingly, wonderingly, comfortingly, knowing...  
148                      [be, be, neat, agreeing, ink]  
149      [entity, descendants, tendency, minute, dint]  

[150 rows x 3 columns]

¿Qué pasa con las demás palabras?

In [34]:
words_encodec_not_unique = []
for i in range(len(words_encodec)):
    word = words_encodec[i][0]
    k_1 = w2v_bert['retrieval_indices'][i]
    if not words_count[word] == 1:
        words_encodec_not_unique.append(words_encodec[i])
        print(f'word: {word}, retrieval: { words_encodec[i][1]}')

word: some, retrieval: ['insist', 'sooner', 'was', 'miss', 'was']
word: this, retrieval: ['the', 'the', 'the', 'the', 'the']
word: fire, retrieval: ['are', 'are', 'are', 'are', 'are']
word: never, retrieval: ['am', 'am', 'am', 'am', 'am']
word: excitement, retrieval: ['nile', 'wonder', 'autumn', 'general', 'twilight']
word: probably, retrieval: ['wildly', 'emperors', 'little', 'solved', 'little']
word: cross, retrieval: ['karl', 'karl', 'hot', 'crowd', 'casus']
word: and, retrieval: ['in', 'in', 'in', 'in', 'in']
word: no, retrieval: ['mode', 'mode', 'mode', 'mode', 'mode']
word: pocket, retrieval: ['back', 'back', 'up', 'up', 'up']
word: voice, retrieval: ['oil', 'war', 'tours', 'holgrave', 'will']
word: an, retrieval: ['in', 'in', 'in', 'in', 'in']
word: keep, retrieval: ['he', 'he', 'he', 'he', 'he']
word: recipe, retrieval: ['be', 'be', 'be', 'be', 'be']
word: that, retrieval: ['the', 'the', 'the', 'the', 'the']
word: for, retrieval: ['were', 'her', 'her', 'her', 'her']
word: durin

In [35]:
words_w2v_not_unique = []
for i in range(len(words_w2v)):
    word = words_w2v[i][0]
    if not words_count[word] == 1:
        words_w2v_not_unique.append(words_w2v[i])
        print(f'word: {word}, retrieval: { words_w2v[i][1]}')

word: cross, retrieval: ['across', 'across', 'across', 'across', 'across']
word: no, retrieval: ['now', 'now', 'now', 'now', 'now']
word: recipe, retrieval: ['stew', 'christie', 'fish', 'he', 'he']
word: drouet, retrieval: ['herbert', 'ardour', 'herbert', 'ardor', 'bread']
word: panada, retrieval: ['canada', 'add', 'atlanta', 'add', 'pleroma']
word: quarters, retrieval: ['quarter', 'quarter', 'quarter', 'corner', 'rope']
word: olden, retrieval: ['old', 'old', 'old', 'old', 'old']
word: call, retrieval: ['called', 'called', 'called', 'called', 'called']
word: venus, retrieval: ['v', 'v', 'main', 'main', 'phoebe']
word: while, retrieval: ['wild', 'wild', 'wild', 'wild', 'wild']
word: edition, retrieval: ['tuition', 'gifted', 'to', 'to', 'distinctions']
word: nor, retrieval: ['or', 'or', 'or', 'or', 'or']
word: whisper, retrieval: ['fervor', 'ardor', 'sward', 'collar', 'ardour']
word: beaten, retrieval: ['beat', 'beat', 'beat', 'beat', 'beat']
word: whom, retrieval: ['who', 'who', 'who', 

In [37]:
import pandas as pd
words_w2v_counter = [item[0] for item in words_w2v_not_unique]
words_encodec_counter = [item[0] for item in words_encodec_not_unique]

# Find the intersection of words in both lists
common_words = set(words_w2v_counter) & set(words_encodec_counter)

# Creating DataFrame with word lists
data = {
    'word': list(common_words),
    'encodec': [next(item[1] for item in words_encodec_not_unique if item[0] == word) for word in common_words],
    'w2v': [next(item[1] for item in words_w2v_not_unique if item[0] == word) for word in common_words]
}

df_not_unique = pd.DataFrame(data)

df_not_unique.to_csv('retrieval_not_unique.csv')

In [44]:
words_w2v_counter = [item[0] for item in words_w2v]
words_encodec_counter = [item[0] for item in words_encodec]

# Find the intersection of words in both lists
w2v_words = set(words_w2v_counter) - set(words_encodec_counter)

# Creating DataFrame with word lists
data = {
    'word': list(w2v_words),
    'w2v': [next(item[1] for item in words_w2v if item[0] == word) for word in w2v_words]
}

df_unique_w2v = pd.DataFrame(data)

df_unique_w2v.to_csv('retrieval_unique_w2v.csv')


In [45]:
words_w2v_counter = [item[0] for item in words_w2v]
words_encodec_counter = [item[0] for item in words_encodec]

# Find the intersection of words in both lists
encodec_words =  set(words_encodec_counter) - set(words_w2v_counter)

# Creating DataFrame with word lists
data = {
    'word': list(encodec_words),
    'encodec': [next(item[1] for item in words_encodec if item[0] == word) for word in encodec_words]
}

df_unique_encodec = pd.DataFrame(data)

df_unique_encodec.to_csv('retrieval_unique_encodec.csv')